Flow.Launcher/MainWindow.xaml.cs

```diff
@@ -72,47 +72,57 @@ private void OnLoaded(object sender, RoutedEventArgs _)
                         {
-                            QueryTextBox.SelectAll();
-                            _viewModel.LastQuerySelected = true;
+                            Activate();
+                            QueryTextBox.Focus();
+                            UpdatePosition();
+                            _settings.ActivateTimes++;
+                            if (!_viewModel.LastQuerySelected)
+                            {
+                                QueryTextBox.SelectAll();
+                                _viewModel.LastQuerySelected = true;
+                            }
+
+                            if (_viewModel.ProgressBarVisibility == Visibility.Visible && isProgressBarStoryboardPaused)
+                            {
+                                _progressBarStoryboard.Begin(ProgressBar);
+                                isProgressBarStoryboardPaused = false;
+                            }
                         }
 
-                        if (_viewModel.ProgressBarVisibility == Visibility.Visible && isProgressBarStoryboardPaused)
+                        if (!isProgressBarStoryboardPaused)
                         {
-                            _progressBarStoryboard.Resume();
-                            isProgressBarStoryboardPaused = false;
+                            _progressBarStoryboard.Stop(ProgressBar);
+                            isProgressBarStoryboardPaused = true;
                         }
+
+                        break;
                     }
-                    else if (!isProgressBarStoryboardPaused)
+                    case nameof(MainViewModel.ProgressBarVisibility):
                     {
-                        _progressBarStoryboard.Pause();
-                        isProgressBarStoryboardPaused = true;
-                    }
-                }
-                else if (e.PropertyName == nameof(MainViewModel.ProgressBarVisibility))
-                {
-                    Dispatcher.Invoke(() =>
-                    {
-                        if (_viewModel.ProgressBarVisibility == Visibility.Hidden && !isProgressBarStoryboardPaused)
+                        Dispatcher.Invoke(async () =>
                         {
-                            _progressBarStoryboard.Pause();
-                            isProgressBarStoryboardPaused = true;
-                        }
-                        else if (_viewModel.MainWindowVisibility == Visibility.Visible && isProgressBarStoryboardPaused)
-                        {
-                            _progressBarStoryboard.Resume();
-                            isProgressBarStoryboardPaused = false;
-                        }
-                    }, System.Windows.Threading.DispatcherPriority.Render);
+                            if (_viewModel.ProgressBarVisibility == Visibility.Hidden && !isProgressBarStoryboardPaused)
+                            {
+                                await Task.Delay(50);
```

Comment:
why the delay?

Reply:
make it stop after hiding, or else user may feel a bit stunt.